File: golang/streaming_pipe_in.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	pipeIn := "input.pipe"

	file, _ := os.Open(pipeIn)
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		fmt.Println(strings.ToUpper(line))
	}
}

File: csharp/StreamingPipeIn.cs
using System;
using System.IO;

class StreamingPipeIn
{
    public static void Main(string[] args)
    {
        using var reader = new StreamReader("input.pipe");

        string line;
        while ((line = reader.ReadLine()) != null)
        {
            Console.WriteLine(line.ToUpper());
        }
    }
}
